frontend/src/metabase/components/EntityMenuItem.jsx

```diff
@@ -0,0 +1,91 @@
+import cxs from 'cxs'
+import React from 'react'
+import { Link } from 'react-router'
+
+import Icon from 'metabase/components/Icon'
+
+const itemClasses = cxs({
+    display: 'flex',
+    alignItems: 'center',
+    cursor: 'pointer',
+    color: '#616D75',
+    paddingLeft: '1.45em',
+    paddingRight: '1.45em',
+    paddingTop: '0.85em',
+    paddingBottom: '0.85em',
+    textDecoration: 'none',
+    transition: 'all 300ms linear',
+    ':hover': {
+        color: '#509ee3'
+    },
+    '> .Icon': {
```

Comment:
Why this isn't part of the `Icon` definition itself? Is it because this allows us to have a css-only implementation for the hover state instead of maintaining a hover state in React components?

The only oddity here is that for `LinkMenuItem` for instance doesn't know anything of its children but its `itemClasses` still style a `.Icon` child. But this isn't a blocker either

Reply:
Not entirely sure I understand the question but I can try to give some context for how this is set up.

Our `currentColor` color inheritance thing that we use for things like `text-brand-hover` works well when the text and icon need to be the same (or you're just using an icon) but since we wanted to fade the icons back a little bit here  it would have taken some serious class gymnastics to get them to look right.

In this particular case it felt like Icon was so crucial to how this entire thing looks and works that explicitly acknowledging that and just styling the icons and how they look when an item is hovered on in the menu here in the component that uses them was the clearest and simplest way to get the visual look we wanted.

Reply:
Yeah I just had never seen this kind of child selector inside a jsx file so I was wondering if this is a pattern that I should use in React components I create as well.

What I meant was that for instance in dashboard list I conditionally applied styles in DashboardListItem depending on whether local `this.state.hover` is true or not.

Reply:
Gotcha. This is actually a feature of `cxs`, which I opted to use here to have the one off styling needs for the menu be co-located with the menu item code.